src/emulator/functionsEmulatorShared.ts

```diff
@@ -128,6 +129,33 @@ export class EmulatedTrigger {
   }
 }
 
+/**
+ * Creates a unique trigger definition for each region a function is defined in.
+ * @param definitions A list of all CloudFunctions in the deployment.
+ * @return A list of all CloudFunctions in the deployment, with copies for each region.
+ */
+export function emulatedFunctionsByRegion(
+  definitions: EmulatedTriggerDefinition[]
+): EmulatedTriggerDefinition[] {
+  const regionDefinitions: EmulatedTriggerDefinition[] = [];
+  for (const def of definitions) {
+    if (!def.regions) {
+      def.regions = ["us-central1"];
+    }
+    // Create a separate CloudFunction for
+    // each region we deploy a function to
+    for (const region of def.regions) {
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+      const defDeepCopy: EmulatedTriggerDefinition = JSON.parse(JSON.stringify(def));
+      defDeepCopy.regions = [region];
+      defDeepCopy.id = `${region}-${defDeepCopy.name}`;
+
+      regionDefinitions.push(defDeepCopy);
+    }
+  }
+  return regionDefinitions;
+}
+
 export function getEmulatedTriggersFromDefinitions(
   definitions: EmulatedTriggerDefinition[],
   module: any
```

Comment:
#### File Comment

When we call `startFunctionRuntime` we use `trigger.name` as the ID which the functions emulator runtime uses. Does this mean that within the runtime we don't distinguish one region from another?

Reply:
Not currently, but that's probably a good idea.  There's a FUNCTION_REGION environment variable that probably should be added. https://cloud.google.com/functions/docs/env-var#nodejs_8_python_37_and_go_111

Reply:
They now distinguish (a triggered function with multiple regions will run once with each region).  However, cloud removed the region env variable, so there's not really anything except logs to tell you which region something was run in.